Core/include/Acts/Geometry/detail/DetectorVolumeUpdators.hpp

```diff
@@ -0,0 +1,148 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Geometry/NavigationDelegates.hpp"
+#include "Acts/Geometry/NavigationState.hpp"
+#include "Acts/Geometry/detail/NavigationStateUpdators.hpp"
+#include "Acts/Utilities/detail/Axis.hpp"
+#include "Acts/Utilities/detail/Grid.hpp"
+
+#include <exception>
+
+namespace Acts {
+namespace Experimental {
+
+class DetectorVolume;
+
+namespace detail {
+
+/// @brief  The end of world sets the volume pointer of the
+/// navigation state to nullptr, usually indicates the end of
+/// the known world, hence the name
+struct EndOfWorldImpl : public IManagedDelegateImpl {
+  /// @brief a null volume link - explicitely
+  ///
+  /// @note the method parameters are ignored
+  inline static void update(const GeometryContext& /*ignored*/,
+                            NavigationState& nState) {
+    nState.currentVolume = nullptr;
+  }
+};
+
+/// @brief Single volume updator, it sets the current navigation
+/// volume to the volume in question
+///
+struct SingleDetectorVolumeImpl : public IManagedDelegateImpl {
+  const DetectorVolume* dVolume = nullptr;
+
+  /// @brief Allowed constructor
+  /// @param sVolume the volume to which it points
+  SingleDetectorVolumeImpl(const DetectorVolume* sVolume) noexcept(false)
+      : dVolume(sVolume) {
+    if (sVolume == nullptr) {
+      throw std::invalid_argument(
+          "DetectorVolumeUpdators: nullptr provided, use EndOfWorld instead.");
+    }
+  }
+
+  SingleDetectorVolumeImpl() = delete;
```

Comment:
could be dropped as it is implicitly deleted

Reply:
Same.